### ISSUE TITLE
Set minimum_os_version project wide?

### ISSUE THREAD

**Issue by keith** [bazelbuild/rules_apple#196]
Currently if you don't set the `minimum_os_version` attribute for `ios_unit_test`, `ios_application` etc, yet you pass `--minimum_os_version 10.0` from the command like, you get this warning:

```
DEBUG: /path/to/log:248:5: The target //:foo does not specify its minimum OS version, so 10.0 from the --ios_minimum_os setting will be used. Please set one for this target specifically by using the minimum_os_version attribute (for example, 'minimum_os_version = "9.0"').
```

Is there a recommended way to set this project wide, so you don't have to add / change this attribute from every target like this?

**Comment by sergiocampama** [comment]
There are a couple of options. 
1. you can create a macro for the rules that you want to have a default minimum os version like so:
```
# my_rules.bzl
load("@build_bazel_rules_apple//apple:ios.bzl", "ios_application")
_DEFAULT_IOS_MIN_OS = "9.0"

def my_ios_application(name, **kwargs):
  minimum_os_version = kwargs.pop("minimum_os_version", _DEFAULT_IOS_MIN_OS)
  ios_application(name = name, minimum_os_version = minimum_os_version, **kwargs)
```

and then in your BUILD file you do:
```
load("//path/to/package:my_rules.bzl", "my_ios_application")

# Replace instances of ios_application with my_ios_application
```

The downside of the macro approach is that when you want to support different behaviors through multiple macros for the same rule it gets harder to use both of them without merging them into one.

2. Add a `tools/.bazelrc` file (relative to the workspace directory) with the following contents:
```
build --ios_minimum_os=<DEFAULT_VALUE>
```

The downside of this is getting that warning for every build. This location will change in the future, see https://github.com/bazelbuild/bazel/issues/4502 for more info.

**Comment by keith** [comment]
Thanks for the detailed answer! A custom rule seems pretty ideal for this case!

**Comment by thomasvl** [comment]
fyi - To avoid the issues if you have more than one default, you can also just use a local bzl and define that they you reuse:

```python
# my_rules.bzl
DEFAULT_IOS_MIN_OS = "9.0"
```
and then:

```python
# BUILD file you do:

load("//path/to/package:my_rules.bzl", "DEFAULT_IOS_MIN_OS")

ios_application(
  "my_app",
  minimum_os_version = DEFAULT_IOS_MIN_OS,
  ...
)
```

**Comment by keith** [comment]
> but I think the main pain point is for test targets

definitely agree

> ...and then you only have to write the default once per package instead of once per target.

This would be nice. Although I'm not sure what project layout you have that would significantly benefit from this case? Our layout currently looks more like this:

```
Modules:
  ModuleA:
    BUILD
  ModuleB:
    BUILD
```

So doing that per package for us would still only cover a single test target (unless there's a way to do use `package` today more globally that I'm not aware of)
